rusty_results/prelude.py

```diff
@@ -368,6 +391,20 @@ def expect_empty(self, msg: str):
     def unwrap_empty(self):
         self.expect_empty("")
 
+    def flatten_one(self) -> "Option[T]":
+        inner: T = self.unwrap()
+        if isinstance(inner, OptionProtocol):
+            return inner
```

Comment:
This is wrong, it would fail on any type where `Self[T]`, `T` is other than `Option`. This could be implemented as something like:

```python
def flatten_one(self) -> "Option[T]":
        if self.map(lambda inner: isinstance(inner, OptionProtocol)).unwrap_or(False):
            return self.unwrap()
        return self
```

Please update the tests to catch this case.